specs/bond-manager.md

```diff
@@ -0,0 +1,112 @@
+in [proposals](./proposals.md)).
+
+The bond manager will need to handle bond logic for a variety of different
+[dispute-games](./dispute-game-interface.md). In the simplest "attestation" dispute game,
+bonds will not be required since the attestors are permissioned. But in more complex games,
+such as the fault dispute game, challengers and defenders perform a series of
+alternating onchain transactions requiring bonds at each step.
+
+## The Bond Problem
+
+At its core, the bond manager is straightforward - it escrows or holds ether and returns
+it at maturity or seized. But the uncertainty of introducing bonds lies in the
+bond _sizing_, i.e. how much should a bond be? Sizing bonds correctly is a function of
+the bond invariant: the bond must be greater than or equal to the cost of the next step.
+If bonds are priced too low, then the bond invariant is violated and there isn't an economic
+incentive to execute the next step. If bonds are priced too high, then the actors posting
+bonds can be priced out.
+
+Below, we outline two different approaches to sizing bonds and the tradeoffs of each.
+
+### Simple Bond
+
+The _Simple Bond_ is a very conservative approach to bond management, establishing a **fixed** bond
+size. The idea behind simple bond pricing is to establish the worst case gas cost for
+the next step in the dispute game.
+
+With this approach, the size of the bond is computed up-front when a dispute game is created.
+For example, in an attestation dispute game, this bond size can be computed using the following
+simple linear function:
+
+```md
+size = number of signers * (gas used to progress game + security overhead)
+```
+
+Notice that since the bond size is linearly proportional to the number of signers, the economic
+security a given bond size provides decreases as the number of signers increases.
+
+Working backwards, if we assume the number of signers to be `5` and the gas used to progress
+the game is `100,000` gas, then the bond size should cover `500,000` gas. This means that a bond
+of `1 ether` would cover the cost of progressing the game for `5` signers as long as the gas price
+(base fee) does not exceed `2,000 gwei` for the entire finalization window. It would be prohibitively
+expensive to keep the settlement layer base fee this high.
+
+### Variable Bond
+
+Better bond heuristics can be used to establish a bond price that accounts for
+the time-weighted gas price. One instance of this called _Varable Bonds_ use a
+separate oracle contract, `GasPriceFluctuationTracker`, that tracks gas fluctuations
+within a pre-determined bounds. This replaces the ideal solution of tracking
+challenge costs over all L1 blocks, but provides a reasonable bounds. The initial
+actors posting this bond are responsible for funding this contract.
+
+## Contract Interface
+
+Below is a minimal interface for the bond manager contract.
+
+```solidity
+interface BondManager {
```

Comment:
I think we need a bit more information to "tag" bonds here. The bond owner is an important piece, but the proposer of an output is not currently tracked in the `OutputProposal` struct: https://github.com/ethereum-optimism/optimism/blob/9c4eeaaa102efe5ed5e4e87d5b3ece9e2d7ca85c/packages/contracts-bedrock/contracts/libraries/Types.sol#L9-L22

Maybe rather than keying bonds by an address, we key them by a unique identifier?

Reply:
i.e., when the `AttestationDisputeGame` resolves, it needs to be able to find the bond associated with the output that it's about to delete. The only information available is the old output root, the l1 timestamp that it was proposed at, and the l2blocknumber that the old output root commits to.

Reply:
Re-worked it a bit :D